### ISSUE TITLE
Documentation of short and long-term goals

### ISSUE THREAD

**Issue by SchokiCoder** [c2lang/documentation#10]
## Long-term
G'day, i propose a (new) page that is listing planned language features to come.
I learned from a comment in "c2c_native/analyser/module_analyser.c2", that templates are planned.
Maybe "documentation/docs/development/current_status.md" could be used for that.
This file has been last changed 6 years ago, is that still up to date?
Instead of giving rough percentages, listing things that need to be done?

## Short-term
"current_status.md" _could_ also be used for short term goals like fixing bug X or implementing feature Y.
Alternatively (my preference) a text file in the repo itself like "todo.txt".
I think a centralized (file per repo) to collect things that need to be done is better.
This way, one can see what problems are on the radar and planned to be fixed or are already fixed.
Problems remaining can be sorted by priority.
I have seen TODO comments in c2c_native that _look_ like they have already been done.

**Comment by bvdberg** [comment]
Yes, that would be a good idea. Templates in the first form (Pure function templates) are already in. The best place to see is test/template/ where are the functionality is tested. I'll put up an extra page with planned stuff indeed.

Which TODOs could be removed? There are always too many, so removing them would be nice.

**Comment by SchokiCoder** [comment]
I tried _and failed_ to find that specific TODO that i found while fixing the Reader.close bug.
But removing that one specific TODO is not the primary goal of this issue.

I rather want some way to organize all known issues and to know what you know.
You know what the most pressing issues are and what features are supposed to be implemented, changed or dropped.
I just think it would increase discoverability of potential tasks  :)

Maybe i can hunt down obsolete TODO's later.

**Comment by bvdberg** [comment]
There are github issues that  we could use, but I feel that TODO's are often much smaller in scope
